src/components/TopbarDesktop/TopbarDesktop.js

```diff
@@ -129,26 +130,27 @@ const TopbarDesktop = props => {
     </NamedLink>
   );
 
-  const listingLink = currentUserListing ? (
-    <NamedLink
-      className={css.createListingLink}
-      name="ListingPage"
-      params={{
-        id: currentUserListing.id.uuid,
-        slug: createSlug(currentUserListing.attributes.title),
-      }}
-    >
-      <span className={css.createListing}>
-        <FormattedMessage id="TopbarDesktop.viewListing" />
-      </span>
-    </NamedLink>
-  ) : (
-    <NamedLink className={css.createListingLink} name="NewListingPage">
-      <span className={css.createListing}>
-        <FormattedMessage id="TopbarDesktop.createListing" />
-      </span>
-    </NamedLink>
-  );
+  const listingLink =
+    authenticatedOnClientSide && currentUserListingFetched && currentUserListing ? (
+      <ListingLink
+        className={css.createListingLink}
+        listing={currentUserListing}
+        children={
+          <span className={css.createListing}>
+            <FormattedMessage id="TopbarDesktop.viewListing" />
+          </span>
+        }
+      />
+    ) : null;
+
+  const createListingLink =
+    !isAuthenticated || (currentUserListingFetched && !currentUserListing) ? (
```

Comment:
It might be better that this follows the same pattern as login and signup links:
```js
const loginLink = isAuthenticatedOrJustHydrated ? null : (...);
```
So, my suggestion is:
```js
const createListingLink = isAuthenticatedOrJustHydrated || (currentUserListingFetched && !currentUserListing) ? null : (...);
```